go/plugins/googlegenai/googlegenai.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package googlegenai
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/genkit"
+	"github.com/firebase/genkit/go/plugins/internal/gemini"
+	"google.golang.org/genai"
+)
+
+const (
+	googleAIProvider = "googleai"
+	vertexAIProvider = "vertexai"
+
+	googleAILabelPrefix = "Google AI"
+	vertexAILabelPrefix = "Vertex AI"
+)
+
+// GoogleAI is a Genkit plugin for interacting with the Google AI service.
+type GoogleAI struct {
+	APIKey string // API key to access the service. If empty, the values of the environment variables GEMINI_API_KEY or GOOGLE_API_KEY will be consulted, in that order.
+
+	gclient *genai.Client // Client for the Google AI service.
+	mu      sync.Mutex    // Mutex to control access.
+	initted bool          // Whether the plugin has been initialized.
+}
+
+// VertexAI is a Genkit plugin for interacting with the Google Vertex AI service.
+type VertexAI struct {
+	ProjectID string // Google Cloud project to use for Vertex AI. If empty, the value of the environment variable GOOGLE_CLOUD_PROJECT will be consulted.
+	Location  string // Location of the Vertex AI service. If empty, GOOGLE_CLOUD_LOCATION and GOOGLE_CLOUD_REGION environment variables will be consulted, in that order.
+
+	gclient *genai.Client // Client for the Vertex AI service.
+	mu      sync.Mutex    // Mutex to control access.
+	initted bool          // Whether the plugin has been initialized.
+}
+
+// Name returns the name of the plugin.
+func (ga *GoogleAI) Name() string {
+	return googleAIProvider
+}
+
+// Name returns the name of the plugin.
+func (v *VertexAI) Name() string {
+	return vertexAIProvider
+}
+
+// Init initializes the Google AI plugin and all known models and embedders.
+// After calling Init, you may call [DefineModel] and [DefineEmbedder] to create
+// and register any additional generative models and embedders
+func (ga *GoogleAI) Init(ctx context.Context, g *genkit.Genkit) (err error) {
+	if ga == nil {
+		ga = &GoogleAI{}
+	}
+	ga.mu.Lock()
+	defer ga.mu.Unlock()
+	if ga.initted {
+		panic(fmt.Sprintf("%s.Init already called", googleAIProvider))
+	}
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("%s.Init: %w", vertexAIProvider, err)
```

Comment:
```suggestion
			err = fmt.Errorf("GoogleAI.Init: %w", err)
```

It should match what's being called which is `GoogleAI.Init()`. There's no benefit to saving the provider in a variable here since we have distinct Inits for both.